components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/DeviceManagementServiceProviderImpl.java

```diff
@@ -596,7 +596,48 @@ public int getDeviceCount() throws DeviceManagementException {
             int deviceCount = this.deviceDAO.getDeviceCount();
             return deviceCount;
         } catch (DeviceManagementDAOException e) {
-            throw new DeviceManagementException("Error occurred while obtaining devices all devices", e);
+            log.error("Error occurred while counting devices", e);
+            throw new DeviceManagementException("Error occurred while counting devices", e);
+        }
+    }
+
+    @Override
+    public List<Device> getDevicesByName(String deviceName, int tenantId) throws DeviceManagementException {
+        List<Device> devicesOfUser = new ArrayList<Device>();
+        List<org.wso2.carbon.device.mgt.core.dto.Device> devicesList;
+        Device convertedDevice;
+        DeviceIdentifier deviceIdentifier;
+        DeviceManager dms;
+        Device dmsDevice;
+        org.wso2.carbon.device.mgt.core.dto.Device device;
+
+        try {
+            devicesList = this.getDeviceDAO().getDevicesByName(deviceName, tenantId);
+        } catch (DeviceManagementDAOException e) {
+            throw new DeviceManagementException("Error occurred while fetching the list of devices that matches to '"
+                                                + deviceName + "'", e);
+        }
+
+        for (int x = 0; x < devicesList.size(); x++) {
+            device = devicesList.get(x);
+            try {
+                device.setDeviceType(deviceTypeDAO.getDeviceType(device.getDeviceTypeId()));
```

Comment:
why do u have to set a device type here? isn't this device object have a type by default at the time of retrieval?